pkg/component/shared/istio.go

```diff
@@ -211,7 +211,7 @@ func AddIstioIngressGateway(
 // In case the zone name is too long the first five characters of the hash of the zone are used as zone identifiers.
 func GetIstioNamespaceForZone(defaultNamespace string, zone string) string {
 	const format = "%s--%s"
-	if ns := fmt.Sprintf(format, defaultNamespace, zone); len(ns) <= validation.DNS1035LabelMaxLength {
+	if ns := fmt.Sprintf(format, defaultNamespace, strings.ToLower(zone)); len(ns) <= validation.DNS1035LabelMaxLength {
```

Comment:
While the fix may work in an environment where always upper case is used this can break in case somebody is as clever as naming the zones with just different casing, e.g. "a" vs. "A".

How about handling any upper case characters like the length overflow case, i.e. use the hash logic below in case any character in the zone name is upper case.

```suggestion
	if ns := fmt.Sprintf(format, defaultNamespace, zone); len(ns) <= validation.DNS1035LabelMaxLength && zone == strings.ToLower(zone) {
```